packages/appeals-service-api/src/routes/responses.js

```diff
@@ -1,11 +1,13 @@
 const express = require('express');
 const {
 	patchResponseByReferenceId,
-	getResponseByReferenceId
+	getResponseByReferenceId,
+	submitQuestionnaireResponse
 } = require('../controllers/responses');
 const router = express.Router();
 
-router.patch('/:journeyId/:referenceId', patchResponseByReferenceId);
+router.patch('/:journeyId/:referenceId/:lpaCode', patchResponseByReferenceId);
```

Comment:
It's primarily driven by the PINS data model and the need to ensure that we pass it on with questionnaire submissions https://pins-ds.atlassian.net.mcas.ms/wiki/spaces/AAPDS/pages/1249312781/LPA+Questionnaire+submission+data, however it means that we can use it for authorization purposes so for LPA users we can validate that the response they are working on belongs to their LPA, which is useful as the appeal references aren't necessarily secret. In terms of storing response for none LPA users, I wouldn't discount storing them in a different table and that at run time we would know the role of the users to determine where we should store their journey responses. We will also know the LPA code for those types of user response submissions as well.

Reply:
In my head it could just be part of the object, not the url, but if you think this will be preferable that's fine. Although I would then ask should the other response api endpoints not also require the lpaCode?

Reply:
In the body we only post the answers we want to update at the moment, everything else cones from the params.

We could refactor that but I don't think there's a great deal of value in that as for other user types we might not care about the LPA code in the future so I don't want to bind that to the payload as it might be more straight forward down the line to say user role x send patch requests here user role sends patch requests there, a bit of a service discovery pattern based on a users role. 

In that respect from a metric and observability point of view  you can monitor requests from different user roles based on the url which is easier to query and observe than a request body.

We shouldn't need the LPA code to query the data on the API because in this instance the LPACode doesn't actually help us query the data, we've got a unique lookup based on the journey id and the appeal reference, in future for other user types we might have a different synthetic id that we use to query journey responses.

Reply:
think it just feels a little off to me as I would normally attempt to make apis at least somewhat restful, so seeing different params for the same resource seems odd, but it's not a blocker and easy enough to change later on as we don't have external consumers